src/org/labkey/test/AssayAPITest.java

```diff
@@ -106,6 +111,51 @@ protected void importAssayAndRun(File assayPath, int pipelineCount, String assay
         assertTextPresent(textToCheck);
     }
 
+    @Test
+    public void testGpatAssayOverAPI() throws Exception
+    {
+        String assayName = "testGpatAssay";
+
+        Connection connection = createDefaultConnection(true);
+        GetProtocolCommand getProtocolCommand = new GetProtocolCommand("General");                      // gets a template from the server
+        ProtocolResponse getProtocolResponse = getProtocolCommand.execute(connection, getCurrentContainerPath());
+
+        Protocol newAssayProtocol = getProtocolResponse.getProtocol();
+        newAssayProtocol.setName(assayName)
+            .allowQCStates(true)
+            .setAllowEditableResults(true)
```

Comment:
I don't think it makes sense to set the "allowQCStates" and "setAllowEditableResults", the assay provider either allows that or does not. For example, if you set those to false for a GPAT assay, I assume the response for the save protocol would still say true for those props.

Reply:
Thanks for this; I implemented most of these not knowing which ones would be ignored or non-settable.